python binding to read and set systemd_default_cgroup

kamalesh-babulal opened the pull request:
This patch series adds a python binding that reads `/var/run/libcgroup/systemd`
and if the file exists, sets the `systemd_default_cgroup`. Then on all the paths
constructed, has the `systemd_default_cgroup` appended to it.  This is used when
cgroup sub-tree is constructed for systemd delegation.  It also adds a test case
to excerise the `cgroup_set_default_systemd_cgroup` API.

drakenclimber commented on tests/ftests/071-sudo-pybindings-set_default_systemd_cgroup.py:
```diff
@@ -0,0 +1,168 @@
+        cause = 'Failed to create %s' % SYSTEMD_DEFAULT_CGROUP_FILE
+        return result, cause
+
+    # create /sys/fs/cgroup/cpu/libcgtests.slice
```
What if this test is run on a cgroup v2 system?  This comment isn't correct I think

kamalesh-babulal replied:
Sure, will replace this in the next version.

drakenclimber commented on tests/ftests/071-sudo-pybindings-set_default_systemd_cgroup.py:
```diff
@@ -0,0 +1,168 @@
+    if result == consts.TEST_FAILED:
+        return result, cause
+
+    # create /sys/fs/cgroup/cpu/test071.scope
```
Ditto.  See comment above

kamalesh-babulal replied:
Sure, will replace this in the next version.

kamalesh-babulal commented on tests/ftests/071-sudo-pybindings-set_default_systemd_cgroup.py:
```diff
@@ -0,0 +1,168 @@
+    # is created by the systemd and we relay on it for checking if the
+    # slice and scope were set as default systemd cgroup (new cgroup root)
+    mode = int(Cgroup.cgroup_mode())
+    if mode == 2:
```
@drakenclimber Thank you so much for the review comments.  The `magical numbers`, will fix them.

kamalesh-babulal commented on tests/ftests/071-sudo-pybindings-set_default_systemd_cgroup.py:
```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-only
+#
+# set_default_systemd_cgroup functionality test using the python bindings
```
Sure, will rename the test case.

kamalesh-babulal commented:
**V2 Changes**
- Addressed review comments from @drakenclimber 
  - Renamed the test case file name as the `{testnumber}-{testsuite}-{testname}.py`
  - Replaced `2` with `Cgroup.Mode`
  - Fixed the `path`  formation of the cgroup in the error path in `create_cgrp()`
  - Fixed the comments

drakenclimber commented:
Pushed to main in commits 67805afc3f02 and 0b4016adcb47.  Pushed to release-3.0 in commits 0cf1f8fde9fb and e167c4847202.  Thanks, @kamalesh-babulal
